Fixed list-s to update schedules before today after the day has passed

saltedfishxx opened the pull request:
- Typing `list-s` after passing 12am today will update schedules today to past schedules.

dishenggg reviewed (changes requested):
I think you can remove all calls to `sort()` in other methods. Sorting here ensures that the list is sorted every time someone gets the schedule list from the addressbook.

saltedfishxx commented:
> I think you can remove all calls to `sort()` in other methods. Sorting here ensures that the list is sorted every time someone gets the schedule list from the addressbook.

Updated in https://github.com/AY2324S1-CS2103T-T17-3/tp/pull/306/commits/e3a3a946157857958c646bd619bdcf983cc8e1b2

dishenggg reviewed (approved):
LGTM!

saltedfishxx commented:
> > I think you can remove all calls to `sort()` in other methods. Sorting here ensures that the list is sorted every time someone gets the schedule list from the addressbook.
> 
> Updated in [e3a3a94](https://github.com/AY2324S1-CS2103T-T17-3/tp/commit/e3a3a946157857958c646bd619bdcf983cc8e1b2)

Failing test cases, reverted one `sort()` call in `add()` method to ensure model returns the correct state of lists. Updated in https://github.com/AY2324S1-CS2103T-T17-3/tp/pull/306/commits/48be313a684e1d273d1ace9a05689f556774d148.

dishenggg commented:
> > > I think you can remove all calls to `sort()` in other methods. Sorting here ensures that the list is sorted every time someone gets the schedule list from the addressbook.
> > 
> > 
> > Updated in [e3a3a94](https://github.com/AY2324S1-CS2103T-T17-3/tp/commit/e3a3a946157857958c646bd619bdcf983cc8e1b2)
> 
> Failing test cases, reverted one `sort()` call in `add()` method to ensure model returns the correct state of lists. Updated in [48be313](https://github.com/AY2324S1-CS2103T-T17-3/tp/commit/48be313a684e1d273d1ace9a05689f556774d148).

Sorry my bad, if this is the case, I think it might just be better to just call `sort()` for all methods modifying the list as well.

mfjkri commented:
I believe if you pull and merge master into this branch, it should stop failing codecov!

mfjkri reviewed (approved):
LGTM
